src/components/CodeManagement/ManageRequestsTab.jsx

```diff
@@ -1,41 +1,70 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { Stack } from '@edx/paragon';
 
+import moment from 'moment';
+import { camelCaseObject } from '@edx/frontend-platform';
 import SubsidyRequestManagementTable, {
   useSubsidyRequests,
   SUPPORTED_SUBSIDY_TYPES,
   PAGE_SIZE,
+  SUBSIDY_REQUEST_STATUS,
 } from '../SubsidyRequestManagementTable';
+import EnterpriseAccessApiService from '../../data/services/EnterpriseAccessApiService';
+import { ApproveCouponCodeRequestModal, DeclineSubsidyRequestModal } from '../subsidy-request-modals';
+import { fetchCouponOrders } from '../../data/actions/coupons';
+import LoadingMessage from '../LoadingMessage';
+import { NoAvailableCodesBanner } from '../subsidy-request-management-alerts';
+
+const ManageRequestsTab = ({
+  enterpriseId, couponsData, loading: loadingCoupons, fetchCoupons,
+}) => {
+  useEffect(() => {
+    fetchCoupons();
+  }, []);
 
-const ManageRequestsTab = ({ enterpriseId }) => {
   const {
     isLoading,
     requests,
     requestsOverview,
     handleFetchRequests,
+    updateRequestStatus,
   } = useSubsidyRequests(enterpriseId, SUPPORTED_SUBSIDY_TYPES.codes);
 
-  /* eslint-disable no-console */
-  const handleApprove = (row) => console.log('approve', row);
-  const handleDecline = (row) => console.log('decline', row);
-  /* eslint-enable no-console */
+  const [selectedRequest, setSelectedRequest] = useState();
+  const [isApproveModalOpen, setIsApproveModalOpen] = useState(false);
+  const [isDenyModalOpen, setIsDenyModalOpen] = useState(false);
+
+  if (loadingCoupons) {
+    return <LoadingMessage className="coupons mt-3" />;
+  }
+
+  const now = moment();
+  const coupons = couponsData.results;
+  const hasAvailableCodes = coupons.some(coupon => moment(coupon.endDate) > now && coupon.numUnassigned > 0);
```

Comment:
nit: This is doing iteration over an array. Should it use `useMemo` so it doesn't need to iterate over `coupons` on each component re-render?